lib/tesla/middleware/telemetry.ex

```diff
@@ -13,21 +13,102 @@ if Code.ensure_loaded?(:telemetry) do
 
     end
 
-    :telemetry.attach("my-tesla-telemetry", [:tesla, :request], fn event, time, meta, config ->
+    :telemetry.attach("my-tesla-telemetry", [:tesla, :request, :stop], fn event, measurements, meta, config ->
       # Do something with the event
     end)
     ```
 
+    ## Telemetry Events
+
+    * `[:tesla, :request, :start]` - emitted at the beginning of the request.
+      * Measurement: `%{time: System.monotonic_time}`
```

Comment:
Start time should **not** be a `monotonic_time`... Monotonic time is needed to accurately calculate durations so it needs to be captured for calculating `duration`, but it can not be converted into an actual timestamp.

Start time needs to be a `System.system_time` otherwise we won't be able to determine the actual beginning of the span

Reply:
I was about to say the same thing :)

Reply:
I wondered if they meant for system time to be taken in the handler and the `time` metadata to be used for duration?

Reply:
@tsloughter - correct. But then, theoretically speaking, we don't need to measure the duration either, because that could also be done in the handler. So where do we draw the line?

Reply:
Maybe we should call these fields "monotonic_time" or hide them altogether, as they are bound to be misused. Thoughts?

Reply:
Definitely should at least name it `monotonic_time` and not just `time`.

Reply:
Moving the discussion to: https://github.com/beam-telemetry/telemetry/issues/57#issuecomment-587668574

Reply:
In the absinthe instrumentation I did, I never put a monotonic time inside the emitted events, just the calculated `duration` and the system time for `start`, all in native time units.

I think it's reasonable to keep the monotonic time handling inside the instrumentation since it's not really obvious how to do it right.

Reply:
I also generated a unique ID for each span and put that on all the events so that it's possible to correlate `start` with it's `stop` / `failure` w/o making the handler do a bunch of manual tracking

Reply:
Why do we need wall clock times this deep in a trace? This is the same behavior we use everywhere. 

https://hexdocs.pm/plug/Plug.Telemetry.html#module-time-unit

Reply:
Yeah, Plug is "incorrect" if anyone is trying to turn that into an actual wall clock start time. 

Monotonic time tracks duration:
```
iex()> start_time = System.monotonic_time
-576460589343512497
iex()> end_time = System.monotonic_time
-576460587151522497
iex()> System.convert_time_unit(end_time - start_time, :native, :millisecond)
2191
```

But has nothing to do with wall clock time:
```
iex()> System.convert_time_unit(System.monotonic_time, :native, :millisecond)
-576460434182
```

The actual start time of a span is critical no matter where it is in the trace, I don't know how you'd visualize it or put the assemble trace back together without it